description/src/description.rs

```diff
@@ -0,0 +1,234 @@
+use anyhow::anyhow;
+use scale_info::{
+    form::PortableForm, Field, PortableRegistry, Type, TypeDef, TypeDefArray, TypeDefBitSequence,
+    TypeDefCompact, TypeDefPrimitive, TypeDefSequence, TypeDefTuple, TypeDefVariant, Variant,
+};
+
+use crate::transformer::Transformer;
+
+use super::formatting::format_type_description;
+
+pub fn type_description(type_id: u32, type_registry: &PortableRegistry) -> anyhow::Result<String> {
+    fn return_type_name_on_recurse(
+        _type_id: u32,
+        ty: &Type<PortableForm>,
+        _transformer: &Transformer<String>,
+    ) -> anyhow::Result<String> {
+        if let Some(type_name) = ty.path.ident() {
+            return Ok(type_name);
+        }
+        Err(anyhow!("Recursive type that did not get handled properly"))
+    }
+
+    let transformer = Transformer::new(
+        ty_description,
+        return_type_name_on_recurse,
+        (),
+        type_registry,
+    );
+    let description = transformer.resolve(type_id)?;
+    let formatted = format_type_description(&description);
+    Ok(formatted)
+}
+
+fn ty_description(
+    _type_id: u32,
+    ty: &Type<PortableForm>,
+    transformer: &Transformer<String>,
+) -> anyhow::Result<String> {
+    let ident = ty.path.ident().unwrap_or_default();
+    let prefix = type_def_prefix(&ty.type_def);
+    let type_def_description = type_def_type_description(&ty.type_def, transformer)?;
+    Ok(format!("{prefix}{ident}{type_def_description}"))
+}
+
+/// todo: clean this up
+fn type_def_prefix(type_def: &TypeDef<PortableForm>) -> &str {
```

Comment:
Create an issue to "clean it up"? 

The implementation doesn't looks complete to me, for instance what about Composite a.k.a structs?  Perhaps I don't understand it properly :)